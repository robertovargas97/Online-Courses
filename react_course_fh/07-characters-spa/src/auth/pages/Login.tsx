import { useContext } from 'react'
import { useNavigate } from 'react-router-dom'
import { AuthContext } from '../context/AuthContext'

export const Login = () => {
  const navigate = useNavigate()
  const { login } = useContext(AuthContext)

  const onLogin = () => {
    const lastPath = localStorage.getItem('lastPath') ?? '/dragon-ball'
    login('Roberto Vargas')
    navigate(lastPath, {
      replace: true,
    })
  }

  const component = (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr />
      <button className="btn btn-success" onClick={onLogin}>
        Login
      </button>
    </div>
  )
  return component
}
