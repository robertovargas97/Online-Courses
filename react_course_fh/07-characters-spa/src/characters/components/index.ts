export * from './CharactersList'
export * from './CharacterCard'
export * from './PageTitle'
