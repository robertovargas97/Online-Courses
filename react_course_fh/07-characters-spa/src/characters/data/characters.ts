export const dragonBall = 'Dragon Ball'
export const jujutsu = 'Jujutsu Kaisen'

export type Character = {
  id: string
  name: string
  anime: string
  power: string
  first_appearance: string
}

export const characters: Character[] = [
  {
    id: 'db-broly',
    name: 'Broly',
    anime: dragonBall,
    power: 'Eraser Cannon',
    first_appearance: 'Dragon Ball Z II',
  },
  {
    id: 'db-buu',
    name: 'Buu',
    anime: dragonBall,
    power: 'Calamity Blaster',
    first_appearance: 'Dragon Ball Z: Majin Buu Saga',
  },
  {
    id: 'db-cell',
    name: 'Cell',
    anime: dragonBall,
    power: 'Absortion',
    first_appearance: 'Dragon Ball Z: Cell Saga',
  },
  {
    id: 'db-frieza',
    name: 'Frieza',
    anime: dragonBall,
    power: 'Death Wave',
    first_appearance: 'Dragon Ball Z: Frieza Saga',
  },
  {
    id: 'db-gogeta',
    name: 'Gogeta',
    anime: dragonBall,
    power: 'Big Bang Attack',
    first_appearance: 'DBZ movie Fusion Reborn',
  },
  {
    id: 'db-gohan',
    name: 'Gohan',
    anime: dragonBall,
    power: 'Masenko',
    first_appearance: 'Dragon Ball Z chapter #196',
  },
  {
    id: 'jujutsu-gojo',
    name: 'Satoru Gojo',
    anime: jujutsu,
    power: 'Infinity ',
    first_appearance: 'Jujutsu Kaisen 0 #1',
  },
  {
    id: 'db-goku',
    name: 'Goku',
    anime: dragonBall,
    power: 'Kamehameha',
    first_appearance: 'The first Dragon Ball chapter',
  },
  {
    id: 'db-piccolo',
    name: 'Piccolo',
    anime: dragonBall,
    power: 'Masenko Beam',
    first_appearance: 'Dragon Ball chapter #161',
  },
  {
    id: 'db-trunks',
    name: 'Trunks',
    anime: dragonBall,
    power: 'Burning Attack',
    first_appearance: 'Dragon Ball chapter #331',
  },
  {
    id: 'db-vegeta',
    name: 'Vegeta',
    anime: dragonBall,
    power: 'Bing Bang Attack',
    first_appearance: 'Dragon Ball chapter #204',
  },
  {
    id: 'jujutsu-hanami',
    name: 'Hanami',
    anime: jujutsu,
    power: 'Roots',
    first_appearance: 'Jujutsu Kaisen chapter #5',
  },
  {
    id: 'jujutsu-itadori',
    name: 'Itadori',
    anime: jujutsu,
    power: 'Black Flash',
    first_appearance: 'Jujutsu Kaisen #1',
  },
  {
    id: 'jujutsu-jogo',
    name: 'Jogo',
    anime: jujutsu,
    power: 'Maximum: Meteor',
    first_appearance: 'Jujutsu Kaisen chapter #5',
  },
  {
    id: 'jujutsu-mahito',
    name: 'Mahito',
    anime: jujutsu,
    power: 'Idle Transfiguration',
    first_appearance: 'Jujutsu Kaisen chapter #9',
  },
  {
    id: 'jujutsu-megumi',
    name: 'Megumi',
    anime: jujutsu,
    power: 'Ten Shadows Technique',
    first_appearance: 'Jujutsu Kaisen #1',
  },
  {
    id: 'jujutsu-nanami',
    name: 'Nanami',
    anime: jujutsu,
    power: 'Overtime',
    first_appearance: 'Jujutsu Kaisen chapter #9',
  },
  {
    id: 'jujutsu-sukuna',
    name: 'Sukuna',
    anime: jujutsu,
    power: 'Dismantle',
    first_appearance: 'Jujutsu Kaisen #1',
  },
  {
    id: 'jujutsu-todo',
    name: 'Todo',
    anime: jujutsu,
    power: 'Boogie Woogie',
    first_appearance: 'Jujutsu Kaisen chapter #8',
  },
]
