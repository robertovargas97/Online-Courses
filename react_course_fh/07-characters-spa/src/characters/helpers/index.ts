export * from './getCharactersByAnime'
export * from './getCharacterById'
export * from './getCharactersByName'
