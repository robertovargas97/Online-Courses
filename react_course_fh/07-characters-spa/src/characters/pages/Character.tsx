export const Character = () => {
  return <div>Hero</div>
}
