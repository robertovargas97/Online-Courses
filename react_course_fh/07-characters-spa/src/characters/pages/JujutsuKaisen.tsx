
export const JujutsuKaisen = () => {
  return (
    <h1>JujutsuKaisen</h1>
  )
}
