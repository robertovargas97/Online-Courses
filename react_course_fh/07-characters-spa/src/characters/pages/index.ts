export * from './DragonBall'
export * from './JujutsuKaisen'