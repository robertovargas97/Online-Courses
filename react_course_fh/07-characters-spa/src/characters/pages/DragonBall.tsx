import { CharactersList } from '../components'
import { dragonBall } from '../data/characters'
import { PageTitle } from '../components/PageTitle'

export const DragonBall = () => {
  const logoUrl =
    'https://www.viniloscasa.com/37253-thickbox/vinilos-y-pegatinas-dragon-ball-z-goku-logo.jpg'

  const component = (
    <>
      <PageTitle title="Dragon Ball" />

      <CharactersList anime={dragonBall} logoUrl={logoUrl} />
    </>
  )
  return component
}
