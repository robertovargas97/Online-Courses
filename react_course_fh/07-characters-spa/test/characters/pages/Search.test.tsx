import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import { MemoryRouter, useNavigate } from 'react-router-dom'
import { Search } from '../../../src/characters/pages'
import userEvent from '@testing-library/user-event'

// Mock the 'react-router-dom' module
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // Use the actual module except for useNavigate
  useNavigate: jest.fn(),
}))

describe('Serch Pages Test Suit', () => {
  beforeEach(() => jest.clearAllMocks())

  test('should display correctly with the default values', () => {
    const { container } = render(
      <MemoryRouter>
        <Search />
      </MemoryRouter>,
    )

    expect(container).toMatchSnapshot()
  })

  test('should display Goku and the value for the queryString', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=Goku']}>
        <Search />
      </MemoryRouter>,
    )

    const input: HTMLInputElement = screen.getByRole('textbox')
    const gokuImages: HTMLImageElement[] = screen.getAllByRole('img')
    const searchAlertDisplay = screen.getByTestId('search-alert').style.display

    expect(input.value).toBe('Goku')
    expect(gokuImages[0].src).toContain('/assets/db-goku.jpg')
    expect(searchAlertDisplay).toBe('none')
  })

  test('should display an error if no character', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=Goku12344']}>
        <Search />
      </MemoryRouter>,
    )
    const errorAlertDisplay = screen.getByTestId('error-alert').style.display
    expect(errorAlertDisplay).not.toBe('none')
  })

  test('should call the navigate to the new screen', async () => {
    // Mock useNavigate
    const mockNavigate = jest.fn()
    ;(useNavigate as jest.Mock).mockReturnValue(mockNavigate)

    render(
      <MemoryRouter initialEntries={['/search']}>
        <Search />
      </MemoryRouter>,
    )

    const input: HTMLInputElement = screen.getByRole('textbox')
    userEvent.type(input, 'Gohan')

    await waitFor(() => {
      expect(input.value).toBe('Gohan')
    })

    const form = screen.getByTestId('search-form')

    fireEvent.submit(form)
    expect(mockNavigate).toHaveBeenCalled()
    expect(mockNavigate).toHaveBeenCalledWith('?query=Gohan')
  })
})
