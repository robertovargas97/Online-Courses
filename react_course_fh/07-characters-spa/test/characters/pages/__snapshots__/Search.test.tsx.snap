// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serch Pages Test Suit should display correctly with the default values 1`] = `
<div>
  <h1>
     Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-4"
    >
      <form
        data-testid="search-form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a character"
          type="text"
          value=""
        />
        <button
          class="btn btn-success mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-8"
    >
      <div
        class="alert alert-info animate__animated animate__fadeIn"
        data-testid="search-alert"
      >
        Search a Hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeIn"
        data-testid="error-alert"
        style="display: none;"
      >
        No results for 
        <b>
           
        </b>
      </div>
    </div>
  </div>
</div>
`;
